interp/src.go

```diff
@@ -26,11 +27,14 @@ func (interp *Interpreter) importSrc(rPath, path string) (string, error) {
 			rPath = "."
 		}
 		dir = filepath.Join(filepath.Dir(interp.Name), rPath, path)
-	} else if dir, rPath, err = pkgDir(interp.context.GOPATH, rPath, path); err != nil {
+	} else if dir, rPath, err = pkgDir(&interp.context, rPath, path); err != nil {
 		return "", err
 	}
 
 	if interp.rdir[path] {
+		// BUG: importing source can trigger panics, which leaves
+		// interp.rdir in a dirty state causing this error to be incorrect
+		// if the package failed to be imported.
```

Comment:
I think what you're saying here is what I've just noticed by testing this PR: with a simple main program, that depends on `github.com/davecgh/go-spew/spew`, a valid go.mod, and GO111MODULE=on, I get a panic. Without this PR, in the same conditions, I would get the error :
`./main.go:4:2: import "github.com/davecgh/go-spew/spew" error: unable to find source related to: "github.com/davecgh/go-spew/spew"`

So I don't think this is an acceptable regression. If this PR does not fully add support for modules (which seems to be the case, given the trivial example I've tried), it should at least handle unsupported cases as gracefully as before.

Reply:
that bug happens whenever a package can not be found, its orthogonal to the package lookup code. it exists today without these changes.

Reply:
releveant issue which I documented this bug over 7 months ago. https://github.com/containous/yaegi/issues/423